src/array/cuda/sddmm.cu

```diff
@@ -91,6 +91,43 @@ void SDDMMCsr(const std::string& op,
   });
 }
 
+/*!
+ * \brief CUDA implementation of g-SDDMM on heterograph using 
+    Csr format.
+ */
+template <int XPU, typename IdType, int bits>
+void SDDMMCsrHetero(const std::string& op,
+              const BcastOff& bcast,
+              const std::vector<CSRMatrix>& vec_csr,
+              std::vector<NDArray> vec_lhs,
+              std::vector<NDArray> vec_rhs,
+              std::vector<NDArray> vec_out,
+              int lhs_target,
+              int rhs_target,
+              const std::vector<dgl_type_t> lhs_eid,
+              const std::vector<dgl_type_t> rhs_eid) {
+  int maxstrm = 16;
+  cudaStream_t stream[maxstrm];
+  for (int i = 0; i < maxstrm; i++)
+    cudaStreamCreate(&(stream[i]));
```

Comment:
These streams should be destroyed at the end of this function, and this call should be wrapped in CUDA_CALL() in case it fails.